### PR TITLE
{Fix #483] add_duration_to_date with leading NA

### LINKED ISSUE #483: add_duration_to_date ungraceful error when durations first element is NA
`add_duration_to_date` yields and ungraceful error when duration has leading `NA`.  For example:

   lubridate:::add_duration_to_date( c(1,NA,2), Sys.Date() )   # OKAY
   lubridate:::add_duration_to_date( c(NA,1,2), Sys.Date() )   # ERROR 
   lubridate:::add_duration_to_date( c(NA,NA), Sys.Date() )   # ERROR

The error given is:

> Error in if (hour(ans) == 0 && minute(ans) == 0 && second(ans) == 0) return(as.Date(ans)) : 
>   missing value where TRUE/FALSE needed

### PR DESCRIPTION
**Pull request by ctbrown**
- Fix code that had wrong test for date in ops-addition.r
- Added tests to test-ops-additions.R

### CODE REVIEW

File: R/ops-addition.r [hunk 1]
```diff
@@ -16,8 +16,16 @@ add_duration_to_date <- function(dur, date) {
   if(is.Date(date)){
     date <- as.POSIXct(date)
     ans <- with_tz(date + dur@.Data, "UTC")
-    if (hour(ans) == 0 && minute(ans) == 0 && second(ans) == 0)
+
+    if ( all(is.na(ans) ) ) return(as.Date(ans))  # ALL NAs
+
+    if ( all( hour(na.omit(ans)) == 0 ) &&
```
Comment (vspinu) [inline R/ops-addition.r]: These 4 `all()` are not needed. Simply doing `all(hour(ans) == 0 & minute(ans) == 0 & second(ans) == 0, na.rm = T)` within that `if` should do the job.

Comment (vspinu) [inline R/ops-addition.r]: Could you also add a changelog entry for this? Thanks.

**Comment by ctbrown** [comment]
Yes. No problem.  Look for an update later today.

C-

**Comment by ctbrown** [comment]
Took longer than expected, but your pull request fixing #483 has been
completed and should be passing all tests.

**Comment by vspinu** [comment]
Thanks. This branch created a rather messy history and you have a small conflicts. Would you care to squash and rebase? I can do it for you later in the week if you don't have time for this.

**Comment by ctbrown** [comment]
No problem.  Probably caused by commits coming in. Will probably provide a
new PR.

**Comment by vspinu** [comment]
> Will probably provide a new PR.

You can simply push with `-force` on this branch.

**Comment by ctbrown** [comment]
​Vitalie,

I think this is properly cleaned up now.  Can you please confirm when you
have a chance.

C-​

**Comment by vspinu** [comment]
It's still conflicting. I will have a look.

File: DESCRIPTION [hunk 2]
```diff
@@ -13,7 +13,8 @@ Authors@R: c(person("Garrett", "Grolemund", role = "aut"),
              person("Doug","Mitarotonda", role="ctb"),
              person("Joseph", "Larmarange", role="ctb"),
              person("Jonathan", "Boiser", role="ctb"),
-             person("Chel Hee", "Lee", role = "ctb"))
+             person("Chel Hee", "Lee", role = "ctb"),
+             person("Christopher", "Brown", role = "ctb" ))
```
Comment (vspinu) [inline DESCRIPTION]: This list includes only most frequent commiters sorted in decreasing order of contributions. The actual number of contributors is very long.

**Comment by vspinu** [comment]
Merged. Thanks!

**Comment by ctbrown** [comment]
Excellent!  Thanks.

C-

**Pull request closed without merging**